pkg/controller/v1alpha1/localmodel/controller.go

```diff
@@ -0,0 +1,658 @@
+}
+
+// Creates a PV and set the localModel as its controller
+func (c *LocalModelReconciler) createPV(spec v1.PersistentVolume, localModel *v1alpha1.ClusterLocalModel) error {
+	persistentVolumes := c.Clientset.CoreV1().PersistentVolumes()
+	if _, err := persistentVolumes.Get(context.TODO(), spec.Name, metav1.GetOptions{}); err != nil {
+		if !apierr.IsNotFound(err) {
+			c.Log.Error(err, "Failed to get PV")
+			return err
+		}
+		c.Log.Info("Create PV", "name", spec.Name)
+		if err := controllerutil.SetControllerReference(localModel, &spec, c.Scheme); err != nil {
+			c.Log.Error(err, "Failed to set controller reference")
+			return err
+		}
+		if _, err := persistentVolumes.Create(context.TODO(), &spec, metav1.CreateOptions{}); err != nil {
+			c.Log.Error(err, "Failed to create PV", "name", spec.Name)
+			return err
+		}
+	}
+	return nil
+}
+
+// Creates a PVC and sets the localModel as its controller
+func (c *LocalModelReconciler) createPVC(spec v1.PersistentVolumeClaim, namespace string, localModel *v1alpha1.ClusterLocalModel) error {
+	persistentVolumeClaims := c.Clientset.CoreV1().PersistentVolumeClaims(namespace)
+	if _, err := persistentVolumeClaims.Get(context.TODO(), spec.Name, metav1.GetOptions{}); err != nil {
+		if !apierr.IsNotFound(err) {
+			c.Log.Error(err, "Failed to get PVC")
+			return err
+		}
+		c.Log.Info("Create PVC", "name", spec.Name, "namespace", namespace)
+		if err := controllerutil.SetControllerReference(localModel, &spec, c.Scheme); err != nil {
+			c.Log.Error(err, "Set controller reference")
+			return err
+		}
+		if _, err := persistentVolumeClaims.Create(context.TODO(), &spec, metav1.CreateOptions{}); err != nil {
+			c.Log.Error(err, "Failed to create PVC", "name", spec.Name)
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *LocalModelReconciler) DownloadModel(ctx context.Context, localModel *v1alpha1api.ClusterLocalModel, nodeGroup *v1alpha1api.LocalModelNodeGroup, pvc v1.PersistentVolumeClaim, jobNamespace string) error {
+	readyNodes, notReadyNodes, err := getNodesFromNodeGroup(nodeGroup, c.Client)
+	if err != nil {
+		return err
+	}
+	c.Log.Info("Downloading to nodes", "node count", len(readyNodes.Items))
+
+	if localModel.Status.NodeStatus == nil {
+		localModel.Status.NodeStatus = make(map[string]v1alpha1api.NodeStatus)
+	}
+	for _, node := range notReadyNodes.Items {
+		if _, ok := localModel.Status.NodeStatus[node.Name]; !ok {
+			localModel.Status.NodeStatus[node.Name] = v1alpha1api.NodeNotReady
+		}
+	}
+	for _, node := range readyNodes.Items {
+		if status, ok := localModel.Status.NodeStatus[node.Name]; ok {
+			if status == v1alpha1api.NodeDownloaded {
+				continue
+			}
+		}
+		jobName := localModel.Name + "-" + node.Name
+		c.Log.Info("Launch download job", "name", jobName)
+		job, err := c.launchDownloadJob(jobName, jobNamespace, localModel, localModel.Spec.SourceModelUri, pvc.Name, node.Name)
+		if err != nil {
+			c.Log.Error(err, "Job error", "name", jobName)
+			return err
+		}
```

Comment:
As long as the job resource is created it is considered successful, it will not be able to track the job creation status for example if it failed to launch.

```
sklearn-iris-kind-control-plane-dd7cp                   0/1     CreateContainerConfigError   0               6m50s
```